src/Pos/Lrc/Arbitrary.hs

```diff
@@ -0,0 +1,73 @@
+-- | Arbitrary instances for Lrc types.
+
+module Pos.Lrc.Arbitrary
+       ( InvalidRichmenStake (..)
+       , ValidRichmenStake (..)
+       ) where
+
+import           Universum
+
+import qualified Data.HashMap.Strict as HM
+import           Pos.Constants       (genesisMpcThd)
+import           Pos.Lrc.Types       (RichmenStake)
+import           Pos.Types.Address   (StakeholderId)
+import           Pos.Types.Core      (Coin, coinPortionDenominator, getCoinPortion,
+                                      mkCoin, unsafeGetCoin)
+import           Test.QuickCheck     (Arbitrary (..), Gen, choose)
+
+-- | Wrapper over 'RichmenStake'. Its 'Arbitrary' instance enforces that the stake
+-- distribution inside must be valid, i.e. all of the coins are non-zero and every
+-- stakeholder has sufficient coins to participate.
```

Comment:
Well, it's valid with respect to `genesisMpcThd`. But there are other thresholds. I see three solutions:
1. Change names to reflect that mpc threshold is taken into account. It's the easiest one.
2. Get rid of these newtype and leave only `genRichmenStake :: CoinPortion -> RichmenStake`. It's easy too, but requires some changes in test suites (which should be easy as well).
3. Pass threshold through reflection (`newtype ValidRichmenStake thd = Valid …`, where `thd` is a phantom type which can be reified into `CoinPortion`). I think it's the best option, but may be harder to do.
Reflection is already used somewhere in code, btw.
3.